library.js

```diff
@@ -412,6 +418,21 @@ plugin.addMiddleware = function(req, res, next) {
 				if (Object.keys(req.cookies).length && req.cookies.hasOwnProperty(plugin.settings.cookieName) && req.cookies[plugin.settings.cookieName].length) {
 					return plugin.process(req.cookies[plugin.settings.cookieName], function(err, uid) {
 						if (err) {
+							if (plugins.hasListeners('filter:sessionSharing.error')) {
+							    return plugins.fireHook('filter:sessionSharing.error', {
+								error: err,
+								uid: uid,
+								res: res,
+								settings: plugin.settings
```

Comment:
I suggest adding `handleGuest` as false here as an indicator that it is in the hook payload available to change

Reply:
The main reason is that we have Child/Adult claims on our website account and child accounts are not allowed to use the forum so we need to handle them. We also set/clear a couple of cookies based on errors. 

Long story, short, I inherited an old cloned copy of Session Sharing (2.1, not even a tag for it on repo) with added custom code. I'm trying to update to the latest SS and pull out the custom stuff into our theme/plugins.

Ideally I don't want to override the whole of session sharing error handling as we are just adding to it in our hook. Would a better way be to move the next/handleGuest calls to under the switch statement and have the hook there? Something like this?

    var handleGuest = false;

    switch (err.message) {
       case 'banned':
           winston.info('[session-sharing] uid ' + uid + ' is banned, not logging them in');
           break;
       case 'payload-invalid':
           winston.warn('[session-sharing] The passed-in payload was invalid and could not be processed');
           break;
       case 'no-match':
           winston.info('[session-sharing] Payload valid, but local account not found.  Assuming guest.');
           handleGuest = true;
           break;
       default:
           winston.warn('[session-sharing] Error encountered while parsing token: ' + err.message);
           break;
    }

    return plugins.fireHook('filter:sessionSharing.error', {
        error: err,
        uid: uid,
        res: res,
        settings: plugin.settings,
        handleGuest: handleGuest 
    }, function(err, data) {
       if (data.handleGuest) {
           return handleGuest.call(null, req, res, next);
       } 

       next();
    });

Reply:
Yes, that'd work as well, will merge that PR :+1: